internal/kafka/internal/services/clusters.go

```diff
@@ -301,6 +301,7 @@ type FindClusterCriteria struct {
 	MultiAZ               bool
 	Status                api.ClusterStatus
 	SupportedInstanceType string
+	ExternalID            string
 }
 
 func (c clusterService) FindCluster(criteria FindClusterCriteria) (*api.Cluster, error) {
```

Comment:
Side comment N.2: This function should be called `FindFirstMatchingCluster` since it is not guaranteed that the passed-in filter will find exactly one cluster.

If OTH, the function's role is always to find exactly one cluster, then the function should check whether the filter returned one result or a list of results instead of blindly returning the first one.

Reply:
I switched to use FindCluster